src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/rest/RestKNNWarmupHandler.java

```diff
@@ -0,0 +1,88 @@
+import com.amazon.opendistroforelasticsearch.knn.plugin.KNNPlugin;
+import com.amazon.opendistroforelasticsearch.knn.plugin.transport.KNNWarmupAction;
+import com.amazon.opendistroforelasticsearch.knn.plugin.transport.KNNWarmupRequest;
+import com.google.common.collect.ImmutableList;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static com.amazon.opendistroforelasticsearch.knn.index.KNNSettings.KNN_INDEX;
+import static org.elasticsearch.action.support.IndicesOptions.strictExpandOpen;
+
+/**
+ * RestHandler for k-NN index warmup API. API provides the ability for a user to load specific indices' k-NN graphs
+ * into memory.
+ */
+public class RestKNNWarmupHandler extends BaseRestHandler {
+    public static String NAME = "knn_warmup_action";
+
+    private IndexNameExpressionResolver indexNameExpressionResolver;
+    private ClusterService clusterService;
+
+    public RestKNNWarmupHandler(Settings settings, RestController controller, ClusterService clusterService,
+                                IndexNameExpressionResolver indexNameExpressionResolver) {
+        this.clusterService = clusterService;
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList.of(
+                new Route(RestRequest.Method.GET, KNNPlugin.KNN_BASE_URI + "/warmup/{index}")
+        );
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+        KNNWarmupRequest knnWarmupRequest = createKNNWarmupRequest(request);
+        return channel -> client.execute(KNNWarmupAction.INSTANCE, knnWarmupRequest, new RestToXContentListener<>(channel));
+    }
+
+    private KNNWarmupRequest createKNNWarmupRequest(RestRequest request) {
+        String[] indexNames = Strings.splitStringByCommaToArray(request.param("index"));
+        Index[] indices =  indexNameExpressionResolver.concreteIndices(clusterService.state(), strictExpandOpen(),
+                indexNames);
+
+        Arrays.stream(indices).forEach(index -> {
+            if (!"true".equals(clusterService.state().metadata().getIndexSafe(index).getSettings().get(KNN_INDEX))) {
+                throw new KNNInvalidIndexException(index.getName(),
+                        "Unable to create warmup index that has 'knn' setting set to false");
```

Comment:
1.  We could update log to tell the exact setting name. How about below message

throw new KNNInvalidIndexException(index.getName(),
                        "Warmup request rejected as one of the index has 'knn' setting set to false");



2. Should we loop through the indices and mention full list of indices that are invalid?

3. How about making ```indexNames``` from String array to ```set``` and pass this as a ```ImmutableSet``` to ```KNNWarmupRequest```. This was we could duplicates.

Reply:
1. Makes sense
2. Yes, I think that would be better design.
3. Yes that makes sense. Will change that.

Reply:
For Point 3, `KNNWarmupRequest` extends `BroadcastRequest` which takes an array of indices. So, I do not think I should modify that. Additionally, `KNNWarmupTransportAction` extends `TransportBroadcastByNodeAction`, which deduplicates the indices [here](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java#L252), so I am not sure change 3 is necessary.

Reply:
For 1, do we want to return the index name that has 'index.knn' setting to be false?

Reply:
Correct. Actually list of index names that have ```index.knn``` setting to false.